### PR TITLE
Chapter 5: Filling in the layout

### PR DESCRIPTION
**Pull request by haitien**
screenshot from 2018-12-28 17-08-37

### CODE REVIEW

**Comment by haitien** [comment]
ready

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (thanhhungchu95) [inline .vscode/settings.json]: đưa file này vào file .gitignore nhé, ko cần đưa file này lên

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
+@import "bootstrap-sprockets";
```
Comment (thanhhungchu95) [inline app/assets/stylesheets/custom.scss]:
```suggestion
@import 'bootstrap-sprockets';
```

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+@import "bootstrap";
```
Comment (thanhhungchu95) [inline app/assets/stylesheets/custom.scss]: same as the suggestion above

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+#logo {
+  float: left;
+  margin-right: 10px;
+  font-size: 1.7em;
+  color: #fff;
```
Comment (thanhhungchu95) [inline app/assets/stylesheets/custom.scss]: give a name for this color

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+p {
+  font-size: 1.1em;
+  line-height: 1.7em;
+}
+
+footer {
+  margin-top: 45px;
+  padding-top: 5px;
+  border-top: 1px solid #eaeaea;
```
Comment (thanhhungchu95) [inline app/assets/stylesheets/custom.scss]: give a name for this color

File: app/views/layouts/_footer.html.erb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      </li>
+      <li>
+        <%= link_to t(".contact"), "#" %>
+      </li>
+      <li>
+        <%= link_to t(".news"), "http://news.railstutorial.org/" %>
+      </li>
+    </ul>
+  </nav>
+</footer>
```
Comment (thanhhungchu95) [inline app/views/layouts/_footer.html.erb]: add an empty line at end of file

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+        <li>
+          <%= link_to t(".home"), "/home" %>
+        </li>
+        <li>
+          <%= link_to t(".help"), "/help" %>
+        </li>
+      </ul>
+    </nav>
+  </div>
+</header>
```
Comment (thanhhungchu95) [inline app/views/layouts/_header.html.erb]: add an empty line at end of file

File: app/views/layouts/_shim.html.erb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<!--[if lt IE 9]>
+  <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/r29/html5.min.js">
+  </script>
+<![endif]-->
```
Comment (thanhhungchu95) [inline app/views/layouts/_shim.html.erb]: add an empty line at end of file

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -1,13 +1,23 @@
...
+  <%= render "layouts/header" %>
+  <div class="container">
     <%= yield %>
-  </body>
-</html>
+    <%= render "layouts/footer" %>
+  </div>
+</body>
+
+</html>
```
Comment (thanhhungchu95) [inline app/views/layouts/application.html.erb]: add an empty line at end of file

File: app/views/static_pages/about.html.erb [hunk 10]
```diff
@@ -1,12 +1,20 @@
...
-  <br><%= t ".intro" %>
+  <br>
+  <%= t ".intro" %>
 </p>
-<%= link_to "En", locale:"en" %>
-<%= link_to "Vi", locale:"vi" %>
+<div class="lang">
+  <%= link_to "En", locale:"en" %>
+  <%= link_to "Vi", locale:"vi" %>
+</div>
```
Comment (thanhhungchu95) [inline app/views/static_pages/about.html.erb]: add an empty line at end of file

File: app/views/static_pages/contact.html.erb [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+  <%= t ".contact_the" %>
+  <%= t ".rort" %>
+  <%= t ".about" %>
+  <%= t ".sample_app" %>
+  <%= link_to t(".contact_page"), "http://www.railstutorial.org/contact" %>
+</p>
+<div class="lang">
+  <%= link_to "En", locale:"en" %>
+  <%= link_to "Vi", locale:"vi" %>
+</div>
```
Comment (thanhhungchu95) [inline app/views/static_pages/contact.html.erb]: add an empty line at end of file

File: app/views/static_pages/help.html.erb [hunk 12]
```diff
@@ -1,10 +1,14 @@
...
   <%= link_to t(".help_section"), "http://www.railstutorial.org/help" %>.
   <%= t ".to_get_help" %>
   <%= link_to t(".rort_book"), "http://www.railstutorial.org/book" %>.
 </p>
-<%= link_to "En", locale:"en" %>
-<%= link_to "Vi", locale:"vi" %>
+<div class="lang">
+  <%= link_to "En", locale:"en" %>
+  <%= link_to "Vi", locale:"vi" %>
+</div>
```
Comment (thanhhungchu95) [inline app/views/static_pages/help.html.erb]: add an empty line at end of file

File: app/views/static_pages/home.html.erb [hunk 13]
```diff
@@ -1,9 +1,19 @@
...
+  </p>
+  <%= link_to t("sign_up"), "#", class: "btn btn-lg btn-primary" %>
+</div>
+<div class="rails-image">
+  <%= link_to image_tag("rails.png", alt: "Rails logo"), "http://rubyonrails.org/" %>
+</div>
+<div class="lang">
+  <%= link_to "En", locale:"en" %>
+  <%= link_to "Vi", locale:"vi" %>
+</div>
```
Comment (thanhhungchu95) [inline app/views/static_pages/home.html.erb]: add an empty line at end of file

File: config/locales/en.yml [hunk 14]
```diff
@@ -20,7 +20,22 @@ en:
...
\ No newline at end of file
+  sign_up: "Sign up now!"
+  layouts:
+    header:
+      home: "Home"
+      help: "Help"
+    footer:
+      about: "About"
+      contact: "Contact"
+      news: "News"
```
Comment (thanhhungchu95) [inline config/locales/en.yml]: add an empty line at end of file

File: config/locales/vi.yml [hunk 15]
```diff
@@ -20,6 +20,23 @@ vi:
...
   ror_sampleapp: "Hướng dẫn ứng dụng mẫu Ruby on Rails"
+  sign_up: "Đăng nhập ngay!"
+  layouts:
+    header:
+      help: "Hỗ trợ"
+    footer:
+      about: "Về chúng tôi"
+      contact: "Liên hệ"
+      news: "Tin tức"
+
```
Comment (thanhhungchu95) [inline config/locales/vi.yml]: remove 2 line

**Comment by thanhhungchu95** [comment]
commented

File: app/assets/stylesheets/custom.scss (hunk 5 above)
Comment (haitien) [reply app/assets/stylesheets/custom.scss]: why we need name of this color while it can represent of hex value?

**Comment by haitien** [comment]
ready

**Comment by haitien** [comment]
ready

File: .gitignore [hunk 16]
```diff
@@ -24,3 +24,4 @@
 
 /config/master.key
 /config/database.yml
+/.vscode
```
Comment (thanhhungchu95) [inline .gitignore]: add an empty line at end of file

File: app/views/layouts/_footer.html.erb [hunk 17]
```diff
@@ -0,0 +1,19 @@
+<footer class="footer">
+  <small>
+    <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+    <a href="http://www.michaelhartl.com/">Michael Hartl</a>
+  </small>
+  <nav>
+    <ul>
+      <li>
+        <%= link_to t(".about"),   "/about" %>
```
Comment (thanhhungchu95) [inline app/views/layouts/_footer.html.erb]: 1 space only

File: app/views/static_pages/about.html.erb [hunk 18]
```diff
@@ -1,12 +1,20 @@
...
   <%= t ".teach" %>
   <%= link_to t("ror"), "http://www.rubyonrails.org/" %>.
-  <br><%= t ".intro" %>
+  <br>
+  <%= t ".intro" %>
 </p>
-<%= link_to "En", locale:"en" %>
-<%= link_to "Vi", locale:"vi" %>
+<div class="lang">
+  <%= link_to "En", locale:"en" %>
```
Comment (thanhhungchu95) [inline app/views/static_pages/about.html.erb]: add a space after the colon

File: app/views/static_pages/contact.html.erb [hunk 19]
```diff
@@ -0,0 +1,13 @@
+<% provide(:title, t(".contact")) %>
+<h1>Contact</h1>
+<p>
+  <%= t ".contact_the" %>
+  <%= t ".rort" %>
+  <%= t ".about" %>
+  <%= t ".sample_app" %>
+  <%= link_to t(".contact_page"), "http://www.railstutorial.org/contact" %>
+</p>
+<div class="lang">
```
Comment (thanhhungchu95) [inline app/views/static_pages/contact.html.erb]: move this section to application layout for reuse

**Comment by haitien** [comment]
ready

File: config/locales/en.yml [hunk 20]
```diff
@@ -20,7 +20,23 @@ en:
...
+
```
Comment (DuongNT1512) [inline config/locales/en.yml]: check lại denfile cho tới khi push lên k có dấu đỏ này thì thôi nhé

File: config/routes.rb [hunk 21]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
-  get "static_pages/home"
-  get "static_pages/help"
-  get "static_pages/about"
-  root "static_pages#home"
+  get '/home', to: 'static_pages#home'
```
Comment (DuongNT1512) [inline config/routes.rb]: double quote

File: .gitignore [hunk 22]
```diff
@@ -22,5 +27,18 @@
...
+
+config/settings.local.yml
+config/settings/*.local.yml
+config/environments/*.local.yml
+
+/config/application.yml
+
+public/javascripts/
+public/uploads/
+app/assets/javascripts/i18n/
```
[the remark this answers is not in the document]
Comment (haitien) [reply .gitignore]: ignore của em hôm qua bị lỗi nên a Hưng bảo em dùng luôn cái ignore của Vương ạ

**Comment by haitien** [comment]
ready

**Comment by DuongNT1512** [comment]
merged

**Pull request merged**